### DISCUSSION TITLE
Could pydantic-core be useful to rust development?

### DISCUSSION THREAD

**Discussion by jimkring** [pydantic/pydantic#4851] [Feature Request]
I'm working on a small, experimental Rust project and using the [serde](https://serde.rs/), [serde_json](https://github.com/serde-rs/json), and [validator](https://docs.rs/validator/latest/validator/#) modules in a way that's similar to what Pydantic offers to Python. It makes me wonder if the new pydantic-core (which is implemented in rust) for Pydantic V2, might have any general utility to the rust community, or if it too domain-specific and coupled to Pydantic/Python. Said differently, I wonder if a pydantic crate for Rust would make sense at all. It's just a random thought, since I'm just getting to know the various rust crates and was searching for "something like pydantic for rust".  Since it appears many python developers are exploring rust, maybe that sort of logical leap is one that others would explore, too.

It's just a thought...

**Comment by ManuelAngel99** [comment]
I'm also interested

**Comment by ShaneMurphy2** [comment]
It would be neat to be able to generate rust struct definitions from Pydantic models

**Comment by samuelcolvin** [comment]
Right now `pydantic-core` wouldn't be any use to rust developers since it's tightly coupled with pyo3 and python.

There's a plan in future to have a compatibility layer that allows pyo3/python to be swapped out for the equivalent bindings for javascript or wasm. If we do that, we could add a third option to swap out both for pure rust types.

Note that:
1. this would likely be slower than serde since the models are not defined at compile time, but rather are dynamic - this could also be useful in some scenarios
2. This is a long way off

I think I wrote this answer somewhere else, but I can find it right now. If anyone else can remember where, a link would be good.
